hoomd/md/mesh/bond.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Mesh Bond potentials."""
+
+from hoomd.md.mesh.potential import MeshPotential
+from hoomd.data.typeparam import TypeParameter
+from hoomd.data.parameterdicts import TypeParameterDict
+
+
+class Harmonic(MeshPotential):
+    r"""Harmonic bond potential.
+
+    :py:class:`Harmonic` specifies a harmonic potential energy between all
+    particles that share an edge within the mesh. For more details see
+    :py:class:`hoomd.md.bond.Harmonic`.
+
+    Args:
+        mesh (:py:mod:`hoomd.mesh.Mesh`): Mesh data structure constraint.
+
+    Attributes:
+        parameter (TypeParameter[dict]):
+            The parameter of the harmonic bonds for the defined mesh.
+            As the mesh can only have one type a type name does not have
+            to be stated. The dictionary has the following keys:
+
+            * ``k`` (`float`, **required**) - potential constant
+              :math:`[\mathrm{energy} \cdot \mathrm{length}^{-2}]`
+
+            * ``r0`` (`float`, **required**) - rest length
+              :math:`[\mathrm{length}]`
+
+    Examples::
+
+        harmonic = mesh.bond.Harmonic(mesh)
+        harmonic.params["mesh"] = dict(k=10.0, r0=1.0)
+    """
+    _cpp_class_name = "PotentialMeshBondHarmonic"
+
+    def __init__(self, mesh):
+        params = TypeParameter("params", "types",
+                               TypeParameterDict(k=float, r0=float, len_keys=1))
+        self._add_typeparam(params)
+
+        super().__init__(mesh)
+
+
+class FENE(MeshPotential):
```

Comment:
```suggestion
class FENEWCA(MeshPotential):
```
This is now `FENEWCA` in `bonds.py`. Rename for consistency.